### PR TITLE
Added support for Ubuntu PPA (chris-lea/redis-server)

### PR DESCRIPTION
**Pull request by adepretis**
+) added manage_repo parameter for class 'redis::install'
+) added tests

### CODE REVIEW

File: manifests/install.pp [hunk 1]
```diff
@@ -14,17 +14,43 @@
...
-      'Debian', 'Ubuntu': {
-        package { 'redis-server' : ensure => $redis_version, }
+      'Ubuntu': {
+        if $manage_repo {
+          class { '::redis::repo': } ->
+          Anchor['redis::repo::begin']
+
+          anchor { 'redis::repo::begin': } ->
+            package { 'redis-server' : ensure => $redis_version, }
+          anchor { 'php::end': }
```
Comment (dwerder) [inline manifests/install.pp]: php::end???

Comment (adepretis) [inline manifests/install.pp]: Sorry - fixed this. The solution was inspired by mayflower-php puppet module; copy/paste error :-(

**Comment by dwerder** [comment]
Thanks for the PR, but why should this module handle repo configs?

It would be a better solution, if you do something like:

```puppet
include '::apt'

::apt::ppa { 'ppa:chris-lea/redis-server': }->
::redis {redis_package => whatever }
```

**Comment by adepretis** [comment]
because many modules bundle repo configs to ensure perfect integration - at least in my personal experience

**Comment by dwerder** [comment]
I am a friend of KISS. And this module installs/configures redis. Managing repos shouild be in a extra module and not bundled. Because to do it right, you have to bundle rhel 5, 6, 7 , debian, ubuntu and so on repos. Thats not part of this module. And users of RHEL for example will handle there repo in another way then Ubuntu users, Or peple have on repo servers in their infrastructure.

I know that it looks easy to have "standalone" modules. But its a pita to maintain and in bigger environment people have their own repo management and will never use that.

**Comment by adepretis** [comment]
I see no problem with just supplying additional repos for specific distributions. This could furthermore streamline the modules behavior when using a repo instead of downloading/building/installing.

For example, when there's only (!) one (default) instance involved, why does the module create a separate initscript (/etc/init.d/redis-server_$name) and config-files (/etc/redis_$name.conf) instead of using the ones bundled with the package (besides that it does not disable the packages default initscript btw.).

These are just a few examples where, right now, the module has some shortcomings. By bundling the repo-management with the module, it could decide by itself (depending on the repo used) which files (configs, initscripts, ...) to manage in which way. That is one of the reasons many other modules do this - at least for some distros this allows for much better integration and less configuration overhead for the ops.

**Comment by adepretis** [comment]
additionally: the module is on GitHub and available for everyone ... Debian, SLES and RHEL ops could easily extend the repos with widely accepted 3rd-party-repos (dotdeb for Debian, whatever is out there for SLES/RHEL, and so on).

**Comment by dwerder** [comment]
Thats true. And someone has to maintain all that. And puppet provides and standard ways to manage your usecase, like a mentioned in the example above. Furthermore is this module mostly designed for complex setup, like multiple redis and sentinel instances on the same machine. And yes, it is also primaly design for people who need a specific and often newest redis version, which for example on redhat meas  download/compile/install. If you just want a plain redis install for Ubuntu with just one basig instance, then there are better modules for that setup like: https://forge.puppetlabs.com/arioch/redis, which does exactly what you want.

**Pull request closed without merging**